app/tournament/tournament.js

```diff
@@ -0,0 +1,93 @@
+import {Player, validatePlayer} from "./player";
+import {Match, ChildMatch} from "./match";
+import * as utils from "../common/utils";
+
+export class Tournament {
+    constructor() {
+        this.players = [];
+        this.matches = [];
+    }
+
+    addPlayer(player) {
+        this.players.push(new Player(player.name, player.rank));
+        sortPlayersByRank(this);
+    }
+
+    addMatch(match) {
+        this.matches.push(match);
+    }
+
+    getRounds() {
+        return this.matches[this.matches.length - 1].round;
+    }
+
+    generate() {
+        generateTournamentBracket(this);
+    }
+
+    validatePlayer(player) {
+        var validationResult = validatePlayer(player);
+
+        if (validationResult.success && this.existsPlayer(player)) {
+            validationResult.error = "Player already enrolled";
+            validationResult.success = false;
```

Comment:
Don't reuse objects. Create new instances, they are cheaper than the errors you may introduce this way.